esp-config/src/generate.rs

```diff
@@ -15,7 +15,7 @@ pub struct ParseError(String);
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum Value {
-    Number(usize),
+    Integer(isize),
```

Comment:
We might want to specify a valid range for these, too

Reply:
I think validation needs a bit more TLC and thought put into it - I'll open a separate issue for that

Reply:
But right now Integer allows for negative number where it doesn't make sense

Reply:
Okay, what happens if we parse a number to a negative (because it's not numeric, but a bitflag for example), and then the crate tries to read it as `usize`? The parser is now complicated enough that I don't want to think about what it does :D

Reply:
Hmm, well that's my friday ruined, thanks! :D

Yeah that could definitely be an issue, perhaps we should split it into signed and unsigned integers :thinking:

Reply:
Sowwy 🥲 parse into i128? :D

Reply:
I think treating these as separate data types is the best move, I'll update the PR.